From: Michael Tokarev <mjt@tls.msk.ru>
Subject: [PATCH] do not fail build if MAXSYMLINKS isn't defined
To: busybox@busybox.net
Date: Mon, 09 Dec 2013 14:45:01 +0400
Message-Id: <1386585901.215548.32434.nullmailer@msgid.tls.msk.ru>
Forwarded: yes

This is needed for, eg, hurd, which is known to have no constraints.

Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>

diff --git a/libbb/xreadlink.c b/libbb/xreadlink.c
index ec95af2..a610de8 100644
--- a/libbb/xreadlink.c
+++ b/libbb/xreadlink.c
@@ -8,6 +8,12 @@
 
 #include "libbb.h"
 
+/* some systems (eg Hurd) does not have MAXSYMLINKS definition,
+ * set it to some reasonable value if it isn't defined */
+#ifndef MAXSYMLINKS
+# define MAXSYMLINKS 20
+#endif
+
 /*
  * NOTE: This function returns a malloced char* that you will have to free
  * yourself.
